src/Avalonia.Controls/Control.cs

```diff
@@ -237,6 +343,17 @@ protected override void OnAttachedToVisualTree(VisualTreeAttachmentEventArgs e)
             InvalidateMirrorTransform();
         }
 
+        /// <inheritdoc/>
+        public override void Render(DrawingContext context)
+        {
+            if (_isLoaded == false)
```

Comment:
This check is to avoid calling OnLoadedCore() unless absolutely needed. Otherwise, it would be called many times for no reason. There is an additional, duplicate check within OnLoadedCore() itself for safety. Both of these checks, while redundant, make sense.